### PR TITLE
Fix replace_x_y with new xr0.15.1

### LINKED ISSUE #22: replace_x_y_nominal_lat_lon should use assign_coords
I found a breaking change with xr0.15.1: http://xarray.pydata.org/en/latest/whats-new.html

```
variable='fgco2'
cat = col.search(experiment_id='esm-piControl', table_id='Omon',
             variable_id=variable)
dset_dict = cat.to_dataset_dict(cdf_kwargs={'chunks': {'time': 12*50}},
                                preprocess=combined_preprocessing)

--> The keys in the returned dictionary of datasets are constructed as follows:
	'activity_id.institution_id.source_id.experiment_id.table_id.grid_label'
                
--> There is/are 6 group(s)

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-76-7beade68e92f> in <module>
      1 dset_dict = cat.to_dataset_dict(cdf_kwargs={'chunks': {'time': 12*50}},
----> 2                                 preprocess=combined_preprocessing)

/srv/conda/envs/notebook/lib/python3.7/site-packages/intake_esm/core.py in to_dataset_dict(self, zarr_kwargs, cdf_kwargs, preprocess, aggregate, storage_options, progressbar)
    372             self.progressbar = progressbar
    373 
--> 374         return self._open_dataset()
    375 
    376     def _open_dataset(self):

/srv/conda/envs/notebook/lib/python3.7/site-packages/intake_esm/core.py in _open_dataset(self)
    484                 progress(futures)
    485 
--> 486             dsets = client.gather(futures)
    487 
    488         self._ds = {group_id: ds for (group_id, ds) in dsets}

/srv/conda/envs/notebook/lib/python3.7/site-packages/distributed/client.py in gather(self, futures, errors, direct, asynchronous)
   1891                 direct=direct,
   1892                 local_worker=local_worker,
-> 1893                 asynchronous=asynchronous,
   1894             )
   1895 

/srv/conda/envs/notebook/lib/python3.7/site-packages/distributed/client.py in sync(self, func, asynchronous, callback_timeout, *args, **kwargs)
    778         else:
    779             return sync(
--> 780                 self.loop, func, *args, callback_timeout=callback_timeout, **kwargs
    781             )
    782 

/srv/conda/envs/notebook/lib/python3.7/site-packages/distributed/utils.py in sync(loop, func, callback_timeout, *args, **kwargs)
    346     if error[0]:
    347         typ, exc, tb = error[0]
--> 348         raise exc.with_traceback(tb)
    349     else:
    350         return result[0]

/srv/conda/envs/notebook/lib/python3.7/site-packages/distributed/utils.py in f()
    330             if callback_timeout is not None:
    331                 future = asyncio.wait_for(future, callback_timeout)
--> 332             result[0] = yield future
    333         except Exception as exc:
    334             error[0] = sys.exc_info()

/srv/conda/envs/notebook/lib/python3.7/site-packages/tornado/gen.py in run(self)
    733 
    734                     try:
--> 735                         value = future.result()
    736                     except Exception:
    737                         exc_info = sys.exc_info()

/srv/conda/envs/notebook/lib/python3.7/site-packages/distributed/client.py in _gather(self, futures, errors, direct, local_worker)
   1750                             exc = CancelledError(key)
   1751                         else:
-> 1752                             raise exception.with_traceback(traceback)
   1753                         raise exc
   1754                     if errors == "skip":

/srv/conda/envs/notebook/lib/python3.7/site-packages/intake_esm/core.py in _load_group_dataset()
    565         zarr_kwargs,
    566         cdf_kwargs,
--> 567         preprocess,
    568     )
    569 

/srv/conda/envs/notebook/lib/python3.7/site-packages/intake_esm/merge_util.py in _aggregate()
    171             return ds
    172 
--> 173     return apply_aggregation(v)
    174 
    175 

/srv/conda/envs/notebook/lib/python3.7/site-packages/intake_esm/merge_util.py in apply_aggregation()
    118             dsets = [
    119                 apply_aggregation(value, agg_column, key=key, level=level + 1)
--> 120                 for key, value in v.items()
    121             ]
    122             keys = list(v.keys())

/srv/conda/envs/notebook/lib/python3.7/site-packages/intake_esm/merge_util.py in <listcomp>()
    118             dsets = [
    119                 apply_aggregation(value, agg_column, key=key, level=level + 1)
--> 120                 for key, value in v.items()
    121             ]
    122             keys = list(v.keys())

/srv/conda/envs/notebook/lib/python3.7/site-packages/intake_esm/merge_util.py in apply_aggregation()
    118             dsets = [
    119                 apply_aggregation(value, agg_column, key=key, level=level + 1)
--> 120                 for key, value in v.items()
    121             ]
    122             keys = list(v.keys())

/srv/conda/envs/notebook/lib/python3.7/site-packages/intake_esm/merge_util.py in <listcomp>()
    118             dsets = [
    119                 apply_aggregation(value, agg_column, key=key, level=level + 1)
--> 120                 for key, value in v.items()
    121             ]
    122             keys = list(v.keys())

/srv/conda/envs/notebook/lib/python3.7/site-packages/intake_esm/merge_util.py in apply_aggregation()
    100                 cdf_kwargs=cdf_kwargs,
    101                 preprocess=preprocess,
--> 102                 varname=varname,
    103             )
    104 

/srv/conda/envs/notebook/lib/python3.7/site-packages/intake_esm/merge_util.py in _open_asset()
    211     else:
    212         logger.debug(f'Applying pre-processing with {preprocess.__name__} function')
--> 213         return preprocess(ds)
    214 
    215 

/srv/conda/envs/notebook/lib/python3.7/site-packages/cmip6_preprocessing/preprocessing.py in combined_preprocessing()
    868         ds = broadcast_lonlat(ds)
    869         # replace x,y with nominal lon,lat
--> 870         ds = replace_x_y_nominal_lat_lon(ds)
    871         # shift all lons to consistent 0-360
    872         ds = correct_lon(ds)

/srv/conda/envs/notebook/lib/python3.7/site-packages/cmip6_preprocessing/preprocessing.py in replace_x_y_nominal_lat_lon()
    773         eq_ind = abs(ds.lat.mean('x')).load().argmin().data
    774         nominal_x = ds.lon.isel(y=eq_ind)
--> 775         ds.coords['x'].data = nominal_x.data
    776         ds.coords['y'].data = nominal_y.data
    777 

/srv/conda/envs/notebook/lib/python3.7/site-packages/xarray/core/common.py in __setattr__()
    260         """
    261         try:
--> 262             object.__setattr__(self, name, value)
    263         except AttributeError as e:
    264             # Don't accidentally shadow custom AttributeErrors, e.g.

/srv/conda/envs/notebook/lib/python3.7/site-packages/xarray/core/dataarray.py in data()
    551     @data.setter
    552     def data(self, value: Any) -> None:
--> 553         self.variable.data = value
    554 
    555     @property

/srv/conda/envs/notebook/lib/python3.7/site-packages/xarray/core/variable.py in data()
   2106     def data(self, data):
   2107         raise ValueError(
-> 2108             f"Cannot assign to the .data attribute of dimension coordinate a.k.a IndexVariable {self.name!r}. "
   2109             f"Please use DataArray.assign_coords, Dataset.assign_coords or Dataset.assign as appropriate."
   2110         )

ValueError: Cannot assign to the .data attribute of dimension coordinate a.k.a IndexVariable 'x'. Please use DataArray.assign_coords, Dataset.assign_coords or Dataset.assign as appropriate.
```

Try out yourself: CMIP6 pangeo cloud
https://binder.pangeo.io/v2/gh/aaronspring/LunchBytes_intake-esm_cloud/try_pangeo-notebooks

### PR DESCRIPTION
**Pull request by aaronspring**
- added test for replace_x_y
- changed replace_x_y

todo:
- no proper assert statement in test
- this is only a run-through test. not really testing for behaviour
- unsure whether this fix works for real CMIP data

Closes #22

### CODE REVIEW

File: cmip6_preprocessing/preprocessing.py [hunk 1]
```diff
@@ -772,8 +772,8 @@ def replace_x_y_nominal_lat_lon(ds):
         # extract the equatorial lat and take those lon values as nominal lon
         eq_ind = abs(ds.lat.mean('x')).load().argmin().data
         nominal_x = ds.lon.isel(y=eq_ind)
-        ds.coords['x'].data = nominal_x.data
-        ds.coords['y'].data = nominal_y.data
+        ds.assign_coords(x=nominal_x.data)
```
Comment (dcherian) [inline cmip6_preprocessing/preprocessing.py]: `ds = ds.assign_coords`?

**Comment by jbusecke** [comment]
Thanks for the quick fix @aaronspring. To keep things organized, can you move the test to the `tests/test_preprocess.py`, and while were at it, could you rename it to `test_preprocessing.py`, to keep the naming coherent with the modules? Otherwise this looks great. Thanks a lot!

**Comment by aaronspring** [comment]
maybe in the future you/we can think about running at least some runthrough tests on each of the preprocess functions.

**Comment by jbusecke** [comment]
> maybe in the future you/we can think about running at least some runthrough tests on each of the preprocess functions.

Absolutely. I had that on my long list, but just havent gotten to it. If you have some ideas and time at your hands, please feel free to add PRs. 

I really want to run these test directly on the cloud data, but have hit some snags so far. If I figure that out, we can extend the tests nicely.

**Comment by jbusecke** [comment]
Seems like you are missing a numpy import.

**Comment by aaronspring** [comment]
> > maybe in the future you/we can think about running at least some runthrough tests on each of the preprocess functions.
> 
> Absolutely. I had that on my long list, but just havent gotten to it. If you have some ideas and time at your hands, please feel free to add PRs.
> 
> I really want to run these test directly on the cloud data, but have hit some snags so far. If I figure that out, we can extend the tests nicely.

I had local tests in mind. Those run fast and locally. However your suggestion would be the needed test for real term use

**Comment by jbusecke** [comment]
> I had local tests in mind. Those run fast and locally. However your suggestion would be the needed test for real term use

I would like to have both. Local tests should be a priority but the tests with the cloud data could ensure that we detect changes here, and hopefully adjust quickly.

**Comment by aaronspring** [comment]
ready to merge from my side

**Comment by jbusecke** [comment]
I am just running it through my 'crawler' notebook checking if this affect loading data in the cloud and then I am ready to merge. You should probably give yourself credit in a `whats_new.rst`?

File: cmip6_preprocessing/preprocessing.py [hunk 2]
```diff
@@ -772,8 +772,8 @@ def replace_x_y_nominal_lat_lon(ds):
         # extract the equatorial lat and take those lon values as nominal lon
         eq_ind = abs(ds.lat.mean('x')).load().argmin().data
         nominal_x = ds.lon.isel(y=eq_ind)
-        ds.coords['x'].data = nominal_x.data
-        ds.coords['y'].data = nominal_y.data
+        ds = ds.assign_coords(x=nominal_x.data)
```
Comment (dcherian) [inline cmip6_preprocessing/preprocessing.py]: could be combined to `ds = ds.assign_coords(x=nominal_x.data, y=nominal_y.data)`

**Comment by aaronspring** [comment]
You can merge this as is

**Comment by jbusecke** [comment]
It seems like there are more instances where the syntax needs to be changed. Sorry for the delayed check on this, I am experiencing my personal dependency hell on this lovely friday.

Do you think we should merge this and I open a new PR or what is the best way for me to add on top of this?

**Comment by dcherian** [comment]
as maintainer you should be able to push to this branch
`gh pr checkout 23` https://cli.github.com/

**Comment by jbusecke** [comment]
Ill merge this and submit my PR on top of it. Just to keep things safe hahaha.

**Comment by jbusecke** [comment]
It seems like the main tests break because a new model was added to the cloud (`AWI-ESM-1-1-LR`). I modified that test to be less strict and also added an empty dict entry in #24.

**Pull request merged**